### PR TITLE
added  Osm visualisation

### PR DESCRIPTION
**Pull request by ErikLille0**
It is still missing a lot, including comments, the removal of debugcode and some fault-recovery

### CODE REVIEW

File: web/server/src/server.cc [hunk 1]
```diff
@@ -121,31 +123,82 @@ int main(int argc, char const** argv) {
...
+      auto const infra_res_dir = infra_dir / osm_file.filename().replace_extension("");
+      exists_or_create_dir(infra_res_dir);
 
-    auto const tmp_dir = infra_res_dir / "tmp";
-    exists_or_create_dir(tmp_dir);
+      auto const osm_server_file = infra_res_dir / osm_file.filename();
 
-    soro::server::import_settings const import_settings(
-        osm_file,
+      // load and copy
```
Comment (julianharbarth) [inline web/server/src/server.cc:169]: Don't think it is necessary to parse the xml file here. If it is faulty it should be caught later, when it is actually used.

File: web/server/src/server.cc [hunk 2]
```diff
@@ -111,6 +111,8 @@ int main(int argc, char const** argv) {
       infra_todo.emplace_back(dir_entry.path());
     }
   }
+  //For debuging purposes only. Technically it only needs to run this once
```
Comment (julianharbarth) [inline web/server/src/server.cc:114]: Don't commit debug code, if possible. Just use -r as a command line option.

File: web/server/src/server.cc [hunk 3]
```diff
@@ -121,31 +123,82 @@ int main(int argc, char const** argv) {
...
-        osm_file,
+      // load and copy
+      pugi::xml_document osm_data;
+      auto const error = osm_data.load_file(osm_file.c_str()); // load
+
+      if (error) {
+          osm_data.save_file(osm_server_file.c_str()); // copy
+      }
+      else {
+          std::cout << "Failed to read real OSM Data. Will resume without it \n";
```
Comment (julianharbarth) [inline web/server/src/server.cc:177]: There should be logging facilities in utl/logging.h
`uLOG(utl::error) << "Failed to read file " << fp.filename();`

File: web/server/src/soro_server.cc [hunk 4]
```diff
@@ -132,8 +134,19 @@ void serve_file(std::string const& decoded_url,
 
 bool serve_tile(server::serve_context& sc, std::string const& decoded_url,
                 auto const& req, auto& res) {
-  static tiles::regex_matcher const matcher{R"(\/(\d+)\/(\d+)\/(\d+).mvt)"};
-  auto const match = matcher.match(decoded_url);
+
```
[the remark this answers is not in the document]
Comment (juweid) [reply web/server/src/soro_server.cc:137]: Im Debug Build von Visual Studio gibt es ohne diese Änderung diesen Fehler beim Auswählen von Infrastruktur in der WebApp
Error

File: web/server/src/server.cc (hunk 1 above)
Comment (julianharbarth) [reply web/server/src/server.cc:148]: Why is parsing necessary to copy a file?
`std::filesystem::copy(from, to)`

File: web/server/src/soro_server.cc (hunk 4 above)
[the remark this answers is not in the document]
Comment (maximilianruesch) [reply web/server/src/soro_server.cc]: This change was reverted as discussed, as is fixed by the updated tiles version.

**Review by julianharbarth — APPROVED** [review]

**Pull request merged**